src/lib/CRSF/CRSF.cpp

```diff
@@ -303,13 +310,28 @@ uint32_t ICACHE_RAM_ATTR CRSF::GetRCdataLastRecv()
 
 void ICACHE_RAM_ATTR CRSF::JustSentRFpacket()
 {
-    CRSF::OpenTXsyncOffset = micros() - CRSF::RCdataLastRecv;
+    // read them in this order to prevent a potential race condition
+    uint32_t last = CRSF::dataLastRecv;
+    uint32_t m = micros();
+    int32_t delta = (int32_t)(m - last);
 
-    if (CRSF::OpenTXsyncOffset > (int32_t)CRSF::RequestedRCpacketInterval) // detect overrun case when the packet arrives too late and caculate negative offsets.
+    if (delta >= (int32_t)CRSF::RequestedRCpacketInterval)
     {
-        CRSF::OpenTXsyncOffset = -(CRSF::OpenTXsyncOffset % CRSF::RequestedRCpacketInterval);
+        // missing/late packet, force resync
+        CRSF::OpenTXsyncOffset = -(delta % CRSF::RequestedRCpacketInterval) * 10;
+        CRSF::OpenTXsyncWindow = 0;
+        CRSF::OpenTXsyncLastSent -= OpenTXsyncPacketInterval;
+#ifdef DEBUG_OPENTX_SYNC
+        DBGLN("Missed packet, forced resync (%d)!", delta);
+#endif
+    }
+    else
+    {
+        // The number of packets in the sync window is how many if in 20ms
```

Comment:
I don't understand this comment. Can you explain the idea of the avg calculation a bit more detailed?

Reply:
Wow, my English escaped me on that comment! 😆